logstash-core/spec/logstash/java_pipeline_spec.rb

```diff
@@ -604,15 +589,9 @@ def flush(options)
       allow(LogStash::Plugin).to receive(:lookup).with("output", "dummyoutput").and_return(::LogStash::Outputs::DummyOutput)
       allow(logger).to receive(:warn)
 
-      # pipeline must be first called outside the thread context because it lazily initialize and will create a
-      # race condition if called in the thread
-      p = pipeline
-      t = Thread.new { p.run }
-      Timeout.timeout(timeout) do
-        sleep(0.1) until pipeline.ready?
-      end
+      pipeline.start
+      sleep(0.01) until pipeline.stopped?
```

Comment:
A hint for the next reader would be helpful.

```suggestion
      # the only input auto-closes, so the pipeline will automatically stop.
      sleep(0.01) until pipeline.stopped?
```